### PR TITLE
Add AppcuesNotificationService package

### PR DESCRIPTION
**Pull request by mmaatttt**
Creates the `AppcuesNotificationService` package and updates the cocoapods example app to have a Notification Service Extension that uses `AppcuesNotificationService` for rich notifications (ie with image).

Cocoapods build 119 is off of this branch.

As the docs/example shows, usage is quite simple:
```swift
// In your Notification Service Extension
import AppcuesNotificationService
class NotificationService: AppcuesNotificationServiceExtension {}
```

(also sc-60926)

## Next
- I'll also add some options for people who need other customization for push (ie something that can be called from within `didReceive`.
- Update the release script to release this new package as well. The version number will be synced to to main SDK version.

### CODE REVIEW

File: Examples/DeveloperCocoapodsExample/NotificationServiceExtension/Info.plist [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>CFBundleName</key>
+	<string>$(PRODUCT_NAME)</string>
+	<key>CFBundleShortVersionString</key>
+	<string>2.0</string>
+	<key>CFBundleVersion</key>
+	<string>3</string>
```
Comment (iujames) [inline Examples/DeveloperCocoapodsExample/NotificationServiceExtension/Info.plist:10]: don't really matter right?

File: Examples/DeveloperCocoapodsExample/NotificationServiceExtension/NotificationService.swift [hunk 2]
```diff
@@ -0,0 +1,10 @@
+//
+//  NotificationService.swift
+//  NotificationServiceExtension
+//
+//  Created by James Ellis on 12/21/23.
```
Comment (iujames) [inline Examples/DeveloperCocoapodsExample/NotificationServiceExtension/NotificationService.swift:5]: hackathon lives!

File: Sources/AppcuesNotificationService/AppcuesNotificationService.docc/AppcuesNotificationService.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# ``AppcuesNotificationService``
+
+Appcues iOS SDK allows you to integrate Appcues experiences into your native iOS and iPadOS apps.
+
+## Overview
+
+![AppcuesKit icon](banner.png)
+
+A Swift library for sending user properties and events to the Appcues API and retrieving and rendering Appcues content based on those properties and events.
```
Comment (iujames) [inline Sources/AppcuesNotificationService/AppcuesNotificationService.docc/AppcuesNotificationService.md:9]: guessing this may be accidental copy

**Review by iujames — APPROVED** [review]
rollin along! couple minor notes

File: Examples/DeveloperCocoapodsExample/NotificationServiceExtension/Info.plist (hunk 1 above)
Comment (mmaatttt) [reply Examples/DeveloperCocoapodsExample/NotificationServiceExtension/Info.plist:10]: You get a little warning
> ⚠️ The CFBundleVersion of an app extension ('1') must match that of its containing parent app ('3').

File: Sources/AppcuesNotificationService/AppcuesNotificationService.docc/AppcuesNotificationService.md (hunk 3 above)
Comment (mmaatttt) [reply Sources/AppcuesNotificationService/AppcuesNotificationService.docc/AppcuesNotificationService.md:9]: Indeed 😅

**Pull request merged**